nuget-to-nix: small fixes, ryujinx: 1.1.213 -> 1.1.223

zhaofengli opened the pull request:
###### Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

#181512 [added a new argument](https://github.com/NixOS/nixpkgs/blob/d61d4e71ba9a8f56e9f2092b7cfa9cffa4253971/pkgs/build-support/dotnet/nuget-to-nix/nuget-to-nix.sh#L13) to `nuget-to-nix`, breaking existing usages in package-specific updater scripts like `./pkgs/applications/emulators/ryujinx/updater.sh`. Let's make it optional to prevent regressions.

Meanwhile, make it possible to override the output file name for `passthru.fetch-deps` so it can be more easily integrated into the package-specific update scripts. As an example, `ryujinx`'s updater script was migrated to `passthru.fetch-deps`.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

IvarWithoutBones commented:
Thanks! I had been meaning to make this possible for quite a while now :)

In my opinion we should merge https://github.com/NixOS/nixpkgs/pull/187070 first seeing how this needs a rebase anyways, it includes some more fixes for Ryujinx.

Afterwards I'll gladly merge this and make an effort in switching the existing update scripts over :)

zhaofengli committed:
nuget-to-nix: Make exclusion file optional

zhaofengli committed:
build-dotnet-module: Allow specifying the output path of fetch-deps

zhaofengli committed:
ryujinx: Force SDL_VIDEODRIVER to x11

Even though it defaults to x11, the user's environment may have
it set to wayland which doesn't current work:

<https://github.com/NixOS/nixpkgs/pull/187070#discussion_r949522925>

zhaofengli commented:
Alternatively, should we have a SDK-specific `nuget-to-nix`, like `dotnetCorePackages.sdk_6_0.passthru.nuget-to-nix` that automatically applies the correct exclusion list?

IvarWithoutBones commented:
> Alternatively, should we have a SDK-specific `nuget-to-nix`, like `dotnetCorePackages.sdk_6_0.passthru.nuget-to-nix` that automatically applies the correct exclusion list?

Since `nuget-to-nix` is just meant to generate lockfiles for `buildDotnetModule`, I dont really see the usecase for that. With `buildDotnetModule` it should be based off of the SDK used by the derivation already.

Im opposed to adding it, I just dont understand why you would need it

IvarWithoutBones commented on pkgs/applications/emulators/ryujinx/updater.sh:
```diff
@@ -75,16 +75,4 @@ fi
-
-popd
-rm -r "$SRC"
+$(nix-build -A ryujinx.fetch-deps --no-out-link) "$DEPS_FILE"
```
Github wont let me comment on the line since its too high above any changed, but https://github.com/NixOS/nixpkgs/pull/187150/files#diff-83a0ea3f9e13f2d6a738a3b9e2bb99b828c78f03af844e9371b762232320cffdR2 should be changed to:
```shell
#! nix-shell -I nixpkgs=./. -i bash -p coreutils gnused curl common-updater-scripts nix-prefetch-git jq
```
The `fetch-deps` script already propagates the `nuget-to-nix` and the SDK.

zhaofengli commented:
Ok, I was asking about the direct usages of `nuget-to-nix` which won't automatically exclude the SDK packages. If we are going to migrate all/most uses to `passthru.fetch-deps` then it shouldn't be a problem.

IvarWithoutBones reviewed (commented):
Oops, appears i missed this before

IvarWithoutBones commented:
> Ok, I was asking about the direct usages of `nuget-to-nix` which won't automatically exclude the SDK packages. If we are going to migrate all/most uses to `passthru.fetch-deps` then it shouldn't be a problem.

Thats the plan :)

Should be a lot more reliable seeing how stuff like the `projectFile`'s and `dotnet{,Restore}Flags` are automatically set with it. Microsoft also recommands to restore in the same environment as you build in, with the same flags as well. `fetch-deps` enforces that as much as possible.

zhaofengli committed:
ryujinx: Migrate updater script to fetch-deps in buildDotnetModule

zhaofengli committed:
ryujinx: 1.1.213 -> 1.1.223

IvarWithoutBones reviewed (approved):
Looks good to me :+1: 

Result of `nixpkgs-review pr 187150` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>nuget-to-nix</li>
    <li>ryujinx</li>
  </ul>
</details>
